codurance/site
# Make Jenkins do a PR preview on a S3 bucket

This is to have a separate site deployment per each PR.
It creates a new S3 bucket per *branch*.
Any subsequent pushes to this branch will result in redeploys, so you can have a fresh site preview any time you want !

Fixes  #418 #519.

How to see the preview:
![2017-01-17-132931_1195x638_scrot](https://cloud.githubusercontent.com/assets/837104/22022291/41b24b7e-dcb9-11e6-8d3a-319be11378cd.png)

As you can see we have now a Jenkins box where it shows the deployment URL, so you can preview the site. Unfortunally for now you have to copy and paste the URL yourself. When we finish the integration with Circle CI it will be fixed as we are going to write a comment instead.

View of the pipeline in the CI:
![2017-01-17-132820_964x470_scrot](https://cloud.githubusercontent.com/assets/837104/22022199/d69dc73c-dcb8-11e6-8df9-7d05484154cc.png)

This PR is to get the preview functionality working, however it does not mean that the work on the CI is done. Next step is to (not done yet)
* Remove the bucket after PR closes 
* Add a bot that auto-closes stale PRs (#512)
* Migrate to CircleCI
